### ISSUE TITLE
Starting server on gcloud fails waiting for HTTP port even when server is up

### ISSUE THREAD

**Issue by nhammond** [StanfordBioinformatics/loom#280]
The playbook fails with this error. This was tested using default https settings for gcloud.

TASK [Wait for Loom server http port to be available] **************************
task path: /Users/s0198839/.loom/playbooks/tasks/start_server.yml:130
Using module file /Users/s0198839/dev/loom-development/venv/lib/python2.7/site-packages/ansible/modules/core/utilities/logic/wait_for.py
<35.184.59.69> ESTABLISH SSH CONNECTION FOR USER: loom
<35.184.59.69> SSH: EXEC ssh -vvv -C -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o 'IdentityFile="/Users/s0198839/.ssh/loom_id_rsa"' -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=loom -o ConnectTimeout=10 -o ControlPath=/Users/s0198839/.ansible/cp/ansible-ssh-%h-%p-%r 35.184.59.69 '/bin/sh -c '"'"'( umask 77 && mkdir -p "` echo $HOME/.ansible/tmp/ansible-tmp-1488220934.61-60392091799962 `" && echo ansible-tmp-1488220934.61-60392091799962="` echo $HOME/.ansible/tmp/ansible-tmp-1488220934.61-60392091799962 `" ) && sleep 0'"'"''
<35.184.59.69> PUT /var/folders/30/xc7q6bkj1f75zp82j4cdy59mc3nhyw/T/tmpAY9mr7 TO /home/loom/.ansible/tmp/ansible-tmp-1488220934.61-60392091799962/wait_for.py
<35.184.59.69> SSH: EXEC sftp -b - -vvv -C -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o 'IdentityFile="/Users/s0198839/.ssh/loom_id_rsa"' -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=loom -o ConnectTimeout=10 -o ControlPath=/Users/s0198839/.ansible/cp/ansible-ssh-%h-%p-%r '[35.184.59.69]'
<35.184.59.69> ESTABLISH SSH CONNECTION FOR USER: loom
<35.184.59.69> SSH: EXEC ssh -vvv -C -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o 'IdentityFile="/Users/s0198839/.ssh/loom_id_rsa"' -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=loom -o ConnectTimeout=10 -o ControlPath=/Users/s0198839/.ansible/cp/ansible-ssh-%h-%p-%r 35.184.59.69 '/bin/sh -c '"'"'chmod u+x /home/loom/.ansible/tmp/ansible-tmp-1488220934.61-60392091799962/ /home/loom/.ansible/tmp/ansible-tmp-1488220934.61-60392091799962/wait_for.py && sleep 0'"'"''
fatal: [35.184.59.69]: UNREACHABLE! => {
    "changed": false, 
    "msg": "Failed to connect to the host via ssh: OpenSSH_6.9p1, LibreSSL 2.1.8\r\ndebug1: Reading configuration data /Users/s0198839/.ssh/config\r\ndebug1: Reading configuration data /etc/ssh/ssh_config\r\ndebug1: /etc/ssh/ssh_config line 21: Applying options for *\r\ndebug1: /etc/ssh/ssh_config line 56: Applying options for *\r\ndebug1: auto-mux: Trying existing master\r\ndebug2: fd 3 setting O_NONBLOCK\r\ndebug2: mux_client_hello_exchange: master version 4\r\ndebug3: mux_client_forwards: request forwardings: 0 local, 0 remote\r\ndebug3: mux_client_request_session: entering\r\ndebug3: mux_client_request_alive: entering\r\ndebug3: mux_client_request_alive: done pid = 27542\r\ndebug3: mux_client_request_session: session request sent\r\ndebug3: mux_client_read_packet: read header failed: Broken pipe\r\nmux_client_request_session: read from master failed: Broken pipe\r\ndebug2: ssh_connect: needpriv 0\r\ndebug1: Connecting to 35.184.59.69 [35.184.59.69] port 22.\r\ndebug2: fd 4 setting O_NONBLOCK\r\ndebug1: connect to address 35.184.59.69 port 22: Operation timed out\r\nssh: connect to host 35.184.59.69 port 22: Operation timed out\r\n", 
    "unreachable": true
}
	to retry, use: --limit @/Users/s0198839/.loom/playbooks/gcloud_start_server.retry

PLAY RECAP *********************************************************************
35.184.59.69               : ok=26   changed=20   unreachable=1    failed=0   
localhost                  : ok=16   changed=8    unreachable=0    failed=0   

ERROR! Playbook to start server failed. There may be active resources such as running docker containers or  VMs that were created by the playbook and not cleaned up.

**Comment by isaacliao** [comment]
Looks like the connection is timing out. Are you able to SSH to that instance using `ssh -i /Users/s0198839/.ssh/loom_id_rsa 35.184.59.69`? If not, could be a firewall rule issue. Is your client machine in the allowable IP range for SSH?

**Comment by nhammond** [comment]
Yes, I can ssh. It has to be as loom user, though: 
ssh -i /Users/s0198839/.ssh/loom_id_rsa loom@35.184.59.69

**Comment by isaacliao** [comment]
Hm, I can't reproduce this. I'm able to start a gcloud server from local machine with no errors using the 0.3.0-prerelease branch.

Possibly relevant settings to check:
```
LOOM_GCLOUD_SERVER_NETWORK: default
LOOM_GCLOUD_CLIENT_USES_SERVER_INTERNAL_IP: False
LOOM_GCLOUD_SERVER_TAGS: https-server
LOOM_GCLOUD_SERVER_EXTERNAL_IP: ephemeral
```

**Comment by nhammond** [comment]
Ok, not a general problem them. Let me take a closer look at my network settings and try to narrow it down.

**Comment by nhammond** [comment]
Failure was caused by "verify server is up" playbook tasks running on server instead of client. Firewall allowed only connections from Stanford, so https requests from self (on Google Cloud VM) failed. Moved these tasks to run on localhost. 29beb9715a4b5f9432f7c9daa7a7af8e71de5da7

**Comment by nhammond** [comment]
Fixed in c6908c2e439d7354af716902543c7c06726c5021
